### PR TITLE
Spring Boot Data JPA init

### PR DESCRIPTION
**Pull request by VArabchuk**

### CODE REVIEW

File: src/main/java/mate/academy/intro/model/Book.java [hunk 1]
```diff
@@ -10,12 +10,16 @@
...
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
 
 @Entity
 @Getter
 @Setter
 @ToString
 @Table(name = "books")
+@SQLDelete(sql = "UPDATE books SET is_deleted = true WHERE id=?")
+@Where(clause = "is_deleted=false")
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/intro/model/Book.java]: check Hibernate version and if Where annotation deprecated or not

File: src/main/java/mate/academy/intro/repository/BookRepository.java [hunk 2]
```diff
@@ -1,13 +1,30 @@
...
+            + "WHERE b.id = :id")
+    void updateBookById(
+            Long id,
+            String title,
+            String author,
+            String isbn,
+            BigDecimal price,
+            String description,
+            String coverImage
+    );
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/intro/repository/BookRepository.java]:
```suggestion
```

File: src/main/java/mate/academy/intro/controller/BookController.java [hunk 3]
```diff
@@ -32,4 +36,16 @@ public BookDto createBook(@RequestBody CreateBookRequestDto createBookRequestDto
...
+
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    @DeleteMapping("/{id}")
+    public void delete(@PathVariable Long id) {
+        bookService.deleteById(id);
+    }
+
+    @ResponseStatus(HttpStatus.OK)
+    @PutMapping("/{id}")
+    void updateBookById(@PathVariable Long id, @RequestBody CreateBookRequestDto requestDto) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/intro/controller/BookController.java]:
```suggestion
    BookDto updateBookById(@PathVariable Long id, @RequestBody CreateBookRequestDto requestDto) {
```

File: src/main/java/mate/academy/intro/service/BookService.java [hunk 4]
```diff
@@ -10,4 +10,8 @@ public interface BookService {
     List<BookDto> getAll();
 
     BookDto getById(Long id);
+
+    void deleteById(Long id);
+
+    void updateById(Long id, CreateBookRequestDto requestDto);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/intro/service/BookService.java]:
```suggestion
    BookDto updateById(Long id, CreateBookRequestDto requestDto);
```

File: src/main/java/mate/academy/intro/service/BookServiceImpl.java [hunk 5]
```diff
@@ -35,4 +35,22 @@ public BookDto getById(Long id) {
...
     }
+
+    @Override
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
+    }
+
+    @Override
+    public void updateById(Long id, CreateBookRequestDto requestDto) {
+        bookRepository.updateBookById(
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/intro/service/BookServiceImpl.java]:
```suggestion
        Book book = bookRepository.findById(id).....
        bookMapper.updateBookFromDto(bookDto, book);
        return bookMapper.toDto(bookRepository.save(book));
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/mate/academy/intro/service/BookServiceImpl.java [hunk 6]
```diff
@@ -16,6 +16,11 @@ public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
     private final BookMapper bookMapper;
 
+    private Book getBookModelById(Long id) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/intro/service/BookServiceImpl.java]: private method should be placed after all public methods

File: src/main/resources/db/changelog/changes/01-create-books-table.yaml [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+                  type: varchar(255)
+              - column:
+                  name: cover_image
+                  type: varchar(255)
+              - column:
+                  name: is_deleted
+                  type: boolean
+                  defaultValueBoolean: false
+                  constraints:
+                    nullable: false
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/changes/01-create-books-table.yaml:48]: check if you have empty line at the end of file

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/resources/db/changelog/changes/01-create-books-table.yaml (hunk 7 above)
Comment (VArabchuk) [reply src/main/resources/db/changelog/changes/01-create-books-table.yaml:48]: There's an empty line in this file

**Review by Elena-Bruyako — APPROVED** [review]

**Pull request merged**